fdbclient/ServerKnobs.cpp

```diff
@@ -921,7 +921,7 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 
 	// encrypt key proxy
 	init( ENABLE_BLOB_GRANULE_COMPRESSION,                     false ); if ( randomize && BUGGIFY ) { ENABLE_BLOB_GRANULE_COMPRESSION = deterministicRandom()->coinflip(); }
-	init( BLOB_GRANULE_COMPRESSION_FILTER,                    "NONE" );
+	init( BLOB_GRANULE_COMPRESSION_FILTER,                    "NONE" ); if ( randomize && BUGGIFY ) { BLOB_GRANULE_COMPRESSION_FILTER = "NONE"; }
```

Comment:
Shouldn't this buggify to ZSTD? I guess we'd have to do that only if ZSTD is enabled though

Reply:
Yeah, it would be tricky at the moment as ZSTD selection when not supported would cause unnecessary assertions. However, unittests does random filter selection based on all supported filters. I think that would work fine for now.

Reply:
Actually couldn't we just change this to pick a random filter from supported filters if it's buggified? That way we'd test everything?

Reply:
Sure, will try to add a `CompressionUtil::getRandomFilter()` API and then try to use it here (hope compilation works, lets see )